common/service-support-cdi/src/main/java/io/helidon/common/servicesupport/cdi/CdiExtensionBase.java

```diff
@@ -0,0 +1,483 @@
+     * @return true if the annotated type should be kept for potential processing later; false otherwise
+     */
+    protected boolean isConcreteNonInterceptor(ProcessAnnotatedType<?> pat) {
+        AnnotatedType<?> annotatedType = pat.getAnnotatedType();
+        Class<?> clazz = annotatedType.getJavaClass();
+
+        // Abstract classes are handled when we deal with a concrete subclass. Also, ignore if @Interceptor is present.
+        if (annotatedType.isAnnotationPresent(Interceptor.class)
+                || Modifier.isAbstract(clazz.getModifiers())) {
+            logger.log(Level.FINER, () -> "Ignoring " + clazz.getName()
+                    + " with annotations " + annotatedType.getAnnotations()
+                    + " for later processing: "
+                    + (Modifier.isAbstract(clazz.getModifiers()) ? "abstract " : "")
+                    + (annotatedType.isAnnotationPresent(Interceptor.class) ? "interceptor " : ""));
+            return false;
+        }
+        logger.log(Level.FINE, () -> "Accepting " + clazz.getName() + " for later bean processing");
+        return true;
+    }
+
+    /**
+     * Make sure the annotated type is neither abstract nor an interceptor and stores the Java class.
+     *
+     * @param pat {@code ProcessAnnotatedType} event
+     * @return true if the annotated type should be kept for potential processing later; false otherwise
+     */
+    protected boolean recordConcreteNonInterceptor(ProcessAnnotatedType<?> pat) {
+        boolean result = isConcreteNonInterceptor(pat);
+        if (result) {
+            annotatedClasses.add(pat.getAnnotatedType().getJavaClass());
+        }
+        return result;
+    }
+
+
+    /**
+     * Records a producer field defined by the application. Ignores producers
+     * with non-default qualifiers and library producers.
+     *
+     * @param ppf Producer field.
+     */
+    protected void recordProducerField(ProcessProducerField<?, ?> ppf) {
+        recordProducerMember("recordProducerField", ppf.getAnnotatedProducerField(), ppf.getBean());
+    }
+
+    /**
+     * Records a producer method defined by the application. Ignores producers
+     * with non-default qualifiers and library producers.
+     *
+     * @param ppm Producer method.
+     */
+    protected void recordProducerMethod(ProcessProducerMethod<?, ?> ppm) {
+        recordProducerMember("recordProducerMethod", ppm.getAnnotatedProducerMethod(), ppm.getBean());
+    }
+
+    protected Map<Bean<?>, AnnotatedMember<?>> producers() {
+        return producers;
+    }
+
+    /**
+     * Registers the service-related endpoint, after security and as CDI initializes the app scope, returning the default
+     * routing for optional use by the caller.
+     *
+     * @param adv app-scoped initialization event
+     * @param bm BeanManager
+     * @param server the ServerCdiExtension
+     * @return default routing
+     */
+    protected Routing.Builder registerService(
```

Comment:
Once again - this is very much server specific. Not all CDI extensions care about webserver or want to have a dependency on `helidon-microprofile-server`. 
A common module should only have dependency on `helidon-microproflie-cdi`.

Reply:
Addressed by renaming/refactoring as described here: https://gist.github.com/tjquinno/d5b5d13d4e1220e6631372c59d2f7d8b